internal/plan/plan.go

```diff
@@ -258,6 +245,93 @@ const (
 	ActionIgnore   ServiceAction = "ignore"
 )
 
+type LogTargets struct {
+	Targets []string
+	Keyword string
+	Replace bool
+}
+
+const (
+	// Valid keywords for LogTargets
+	DefaultLogTargets = "default"
+	AllLogTargets     = "all"
+	NoLogTargets      = "none"
+
+	// Tag used to replace value (rather than append)
+	logTargetsReplaceTag = "!replace"
+)
+
+func (t *LogTargets) UnmarshalYAML(value *yaml.Node) error {
+	// Check for !replace tag
+	t.Replace = (value.Tag == logTargetsReplaceTag)
```

Comment:
I think we should disallow `!replace` when using a keyword, because it doesn't really add any meaning over `none` when using something like `!replace none`, right? The user might be confused and try the latter, so I think we should disallow it to avoid that confusion. So the field spec would be:

```
log-targets: default | all | none | [!replace] <list of targets>
```

Reply:
"Disallow" is a strong word. Do you mean the following should result in a parsing error?
```yaml
services:
  svc1:
    command: foo
    override: merge
    log-targets: !replace none
```

Because currently, you are allowed to put whatever tags you want on any field, and Pebble doesn't care. This parses just fine:
```yaml
services:
  svc1:
    command: !mycustomtag foo
    override: !iliketagging merge
```

Reply:
Ah yes, parsing error is what I meant. However, in light of what you've said about arbitrary tags being allowed in general, I retract that -- resolving.